Proposed changes to Gherkin scenarios

radinamatic opened the pull request:
### Summary
Ended up being a bit more changes that I intended :blush: 

### Reviewer guidance
Read through and see if it still covers all that you intended to originally.

### References
…

----

### Contributor Checklist


PR process:

- [ ] PR has the correct target branch and milestone
- [ ] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

Testing:

- [ ] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

jredrejo commented on integration_testing/features/admin/admin-bulk-export-users-classrooms.feature:
```diff
@@ -7,8 +7,9 @@ Feature: Admin bulk export of users and classrooms
-      And users belong only to my facility, even if there are more facilities in the device
+      And I am able to open or save or the file 'users_<date>_<time>.csv'
+      # <date> is current date and <time> current time, for example users_20200420_194415.csv
+    When I open the CSV file
```
ah, great I didn't know I could begin with another `When` in the same scenario

radinamatic replied:
It might not be completely kosher if we were using this in automated BDD testing, but since these are steps for manual testing, we do use several `When > Then` pairs in the same scenario.

jredrejo commented on integration_testing/features/admin/admin-bulk-import-users-classrooms.feature:
```diff
@@ -5,32 +5,35 @@ Feature: Admin bulk export of users and classrooms
 
-  Scenario: See CSV file format needed to import users
-    When I click on "Import" button under *Import and export users* heading
+  Scenario: See CSV the file format requirements
```
I don't understand the meaning of this `the`

radinamatic replied:
My bad, it should have said `See the CSV file format requirements` :woman_facepalming: .

jredrejo commented on integration_testing/features/admin/admin-bulk-export-users-classrooms.feature:
```diff
@@ -7,8 +7,9 @@ Feature: Admin bulk export of users and classrooms
 
   Scenario: Export a CSV file containing all users in the facility
-    When I click on "Export" button under *Import and export users* heading
+    When I click on *Export* button under *Import and export users* heading
```
good to know *  has to be used always, I copied this format from a different feature file, so it seems we have some incosistency in them.

radinamatic replied:
It's not a big deal, but we started early on with the convention to enclose **all UI elements** inside the *`*`* chars, and not capitalize all letters even though they do appear capitalized in the Kolibri UI. 

https://github.com/learningequality/kolibri/blob/develop/integration_testing/features/README.md#ui-elements-conventions

I try to edit them for consistency when people contribute Gherkin scenarios, but might have missed some instances.

jredrejo replied:
:+1:

jredrejo commented on integration_testing/features/super-admin/super-admin-bulk-export-users.feature:
```diff
@@ -3,18 +3,24 @@ Feature: Super Admin export users for a facility in a device with manage bulkexp
+    Then I see the 'users_<date>_<time>.csv' file # <date> is current date and <time> current time, for example users_20200420_194415.csv
+    When I open the file with a text editor or a spreadsheet application
+      And I see one row per user in the current facility
+      And each row lists user info and classes they are enrolled or assigned (if they are coach)
```
I miss an `a` before coach

radinamatic replied:
You are correct, it would sound better with it.

jredrejo commented on integration_testing/features/super-admin/super-admin-bulk-import-users.feature:
```diff
@@ -1,28 +1,35 @@
-Feature: Super Admin import users into a facility in a device with manage bulkexportusers command
-    Super Admin needs to be able to do a bulk import of the users and classes, with the information to enroll or assign coaches to the classes
+Feature: Super Admin bulk import users into a facility
+    Super Admin needs to be able to do a bulk import of users and classes using the command line
```
I think that stating that the enroll or assign information is important. This is the most important difference between this command and the existing import users command.

radinamatic replied:
Good point, wasn't thinking about that aspect, will add back! :+1:

jredrejo commented on integration_testing/features/super-admin/super-admin-bulk-import-users.feature:
```diff
@@ -1,28 +1,35 @@
+    Then I see a report containing the number of users and classes to be created, deleted and updated
+      And I see the list of errors, if any
+    When I open the facility in the browser
+      And I can see the new users and classes have been created, with the coaches assigned and the learners enrolled
```
shouldn't it be `Then` instead of `And`?

radinamatic replied:
You got me! :bowing_woman:

jredrejo commented:
@radinamatic I have left a few comments in the PR, very few because I trust more your gherking knowledge than mine, so in case of doubt you're the right one :)

radinamatic commented:
No worries, as you can see, second pair of :eyes: *over the second pair of* :eyes: is always a good thing. Thanks for the thorough reading! :+1: 

I trust my latest commit addressed all your comments! :slightly_smiling_face:
